### ISSUE TITLE
Audio lagging

### ISSUE THREAD

**Issue by stepanzak** [Pithaya/spicetify-apps-dist#3]
Hi, when I play a song with the eternal jukebox website, everything is fine.
However, when I use this app, I can clearly hear audio lagging everytime it's skipping to another part of the song.
That makes this custom app unusable for me and for most people I think.

**Comment by urthurao** [comment]
same for me. songs just sound terrible. did the dev even test it before releasing it?

**Comment by Pithaya** [comment]
Yes there are still issues with the sound skips. This is one of the main reasons the custom app is tagged as "beta" on the marketplace, it's still a work in progress. I guess it would be better to have some sort of disclaimer on the readme as well.
Anyway, the issue here is that while the eternal jukebox website has direct access to the audio that's playing, there's no way to have the same level of control through spicetify, so unfortunately I think some lag can't be avoided. There's probably some way to mitigate it so it's not as bad as it is now though.

**Comment by KawaiiiSheep** [comment]
any update or fix for it? all good if not

**Comment by Pithaya** [comment]
Not yet, I tried a few things but it seems like there's always some lag when controlling the player through spicetify.
What I'm probably going to do is try to add a "mode switch" option to the settings, that would have either a "spotify" mode (the way the app is currently doing, controlling the song through spicetify with lags and all), and a "youtube" mode that will work the same way that the eternal jukebox website does it: find the song on youtube, download the audio, and play that directly. 
Clearly not an ideal solution, but I don't really see another way to fix it

**Comment by Pithaya** [comment]
I'm still working on this. Downloading the audio from youtube is going to be too hard to do with the limitations of the custom app after all, so I'm trying other things.
I'm pretty sure I'll have something working eventually, it's still going to take a little time though.

**Comment by sneazy-ibo** [comment]
Any updates?

**Comment by jxxxxst** [comment]
This extension was just released. Maybe it's what you are looking for @Pithaya 
https://github.com/FoxRefire/spiceDL

**Comment by MrDomino4195** [comment]
Any updates on this?

**Comment by Flanpit** [comment]
Any updates? :3

**Comment by mountaincastle47** [comment]
update?

**Comment by Pithaya** [comment]
Sorry for not being more active on this! I've tried a few things and in the end I'm not sure there's a lot I can do to control audio directly without making users install stuff outside of spotify... which would be really annoying.
I think the part of the app that handles the looping really needs a rewrite after all this time, so I'll have to see what I can do about the issue once it's done.
